optax/_src/loss.py

```diff
@@ -358,3 +358,33 @@ def loop_body(prev, x):
   per_seq_loss = -jnp.einsum('bn,bn->b', logalpha_phi_last, one_hot)
 
   return per_seq_loss
+
+
+def kl_div_loss(predictions: chex.Array,
+                targets: chex.Array,
+                log_targets: bool = False) -> chex.Array:
```

Comment:
I know this is how pytorch does it but since we only have this one function (as opposed to pytorch) I think we should avoid the boolean flag by having two functions: `kl_div_loss` and `kl_div_loss_with_log_targets`.

Aside from the usual arguments against boolean flags, there are two more arguments against having it as a flag in this case in my opinion: the boolean flag changes the meaning of the other parameter (from targets to log_targets) without changing the name which makes the resulting code harder to read and debug. Furthermore, reading from left to right, the meaning of targets will become clear only later at the end of the line or only in the line below.

That way we can also use the name log_predictions (and targets/log_targets) as names for the function inputs without causing confusion with the boolean flag.